evdev/uinput.c

```diff
@@ -97,6 +97,24 @@ uinput_set_prop(PyObject *self, PyObject *args)
         return NULL;
 }
 
+static PyObject *
+uinput_get_sysname(PyObject *self, PyObject *args)
+{
+    int fd;
+    char sysname[64];
+
+    int ret = PyArg_ParseTuple(args, "i", &fd);
+    if (!ret) return NULL;
+
+    if (ioctl(fd, UI_GET_SYSNAME(sizeof(sysname)), &sysname) < 0)
+        goto on_err;
```

Comment:
I can see that the other code in this file also does that. But I was wondering, what is the downside here to doing 

```c
    if (ioctl(fd, UI_GET_SYSNAME(sizeof(sysname)), &sysname) < 0) {
        PyErr_SetFromErrno(PyExc_IOError);
        return NULL;
    }
```

?

goto does seem to be somewhat acceptable for error handling apparently, but still isn't very popular (https://stackoverflow.com/questions/788903/valid-use-of-goto-for-error-management-in-c)

Reply:
I don't think there is any downside to doing this. In case the function makes multiple syscalls which can error, it makes sense to have a separate `on_err` section to deduplicate the error handling logic. In this case, the function only has a single point at which an IOError happens, so there's no real advantage.

In the end, the only reason I added the explicit `on_err` section is because that is what most of the surrounding code did, and when contributing code I think it is best to just follow the style of the surrounding code: "let's change the error handling guidelines" is a separate issue from "how to find the device node corresponding to uinput devices."

But in case you're wondering, I can think of some arguments in favor of the explicit `goto on_err`:
* For style purposes the explicit `goto on_err` is similar to defining a function that gets called from only one place: it doesn't affect the result of the code, but does tell you the intent of those lines of code.
* You may want to make functions that have only a single point of failure have the same structure as functions that have multiple points of failure: it makes pattern recognition easier for a person reading the file;
* If you were to use the policy "inline error handling if there is only one point of failure", then you would have to refactor the error handling if you were to add or remove a second point of failure.

Reply:
Thanks